src/sentry/integrations/source_code_management/repository.py

```diff
@@ -107,34 +113,31 @@ def check_file(self, repo: Repository, filepath: str, branch: str | None = None)
         filepath: file from the stacktrace (string)
         branch: commitsha or default_branch (string)
         """
-        filepath = filepath.lstrip("/")
-        try:
-            client = self.get_client()
-        except (Identity.DoesNotExist, IntegrationError):
-            sentry_sdk.capture_exception()
-            return None
-        try:
-            response = client.check_file(repo, filepath, branch)
-            metrics.incr(
-                REPOSITORY_INTEGRATION_CHECK_FILE_METRIC.format(result="success"),
-                tags={"integration": self.integration_name},
-            )
-            if not response:
-                return None
-        except IdentityNotValid:
-            return None
-        except ApiError as e:
-            if e.code != 404:
-                metrics.incr(
-                    REPOSITORY_INTEGRATION_CHECK_FILE_METRIC.format(result="failure"),
-                    tags={"integration": self.integration_name},
-                )
+        with self.record_event(
+            RepositoryIntegrationInteractionType.CHECK_FILE
+        ).capture() as lifecycle:
+            filepath = filepath.lstrip("/")
+            try:
+                client = self.get_client()
+            except (Identity.DoesNotExist, IntegrationError):
                 sentry_sdk.capture_exception()
-                raise
+                return None
+            try:
+                response = client.check_file(repo, filepath, branch)
+                if not response:
+                    return None
+            except IdentityNotValid:
+                lifecycle.record_failure(IdentityNotValid)
+                return None
+            except ApiError as e:
+                if e.code != 404:
+                    sentry_sdk.capture_exception()
+                    raise
 
-            return None
+                lifecycle.record_failure(ApiError)
```

Comment:
we shouldn't need to fail here, the raise should cover non-404 errors. anything that returns `None` seems to be an acceptable error (could probably ask issues team)